traffic_ops/traffic_ops_golang/cdn/cdns.go

```diff
@@ -20,71 +20,324 @@ package cdn
+		api.HandleErr(w, r, tx.Tx, http.StatusBadRequest, util.JoinErrs(errs), nil)
+		return
+	}
+
+	if inf.Config.UseIMS {
+		runSecond, maxTime = ims.TryIfModifiedSinceQuery(tx, r.Header, queryValues, SelectMaxLastUpdatedQuery(where))
+		if !runSecond {
+			log.Debugln("IMS HIT")
+			api.AddLastModifiedHdr(w, maxTime)
+			w.WriteHeader(http.StatusNotModified)
+			return
+		}
+		log.Debugln("IMS MISS")
+	} else {
+		log.Debugln("Non IMS request")
+	}
+
+	query := selectQuery(inf.Version) + where + orderBy + pagination
+	rows, err := tx.NamedQuery(query, queryValues)
+	if err != nil {
+		api.HandleErr(w, r, tx.Tx, http.StatusNotFound, nil, fmt.Errorf("cdn get: error getting cdn(s): %w", err))
+		return
+	}
+	defer log.Close(rows, "unable to close DB connection")
+
+	cdn := tc.CDNV5{}
+	cdns := []tc.CDNV5{}
+	for rows.Next() {
+		if err = rows.Scan(&cdn.DNSSECEnabled, &cdn.DomainName, &cdn.ID, &cdn.LastUpdated, &cdn.TTLOverride, &cdn.Name); err != nil {
+			api.HandleErr(w, r, tx.Tx, http.StatusInternalServerError, nil, fmt.Errorf("error getting cdn(s): %w", err))
+			return
+		}
+		cdns = append(cdns, cdn)
+	}
+
+	api.WriteResp(w, r, cdns)
+	return
+}
+
+func Create(w http.ResponseWriter, r *http.Request) {
+	cdn := tc.CDNV5{}
+
+	inf, userErr, sysErr, errCode := api.NewInfo(r, nil, nil)
+	if userErr != nil || sysErr != nil {
+		api.HandleErr(w, r, inf.Tx.Tx, errCode, userErr, sysErr)
+		return
+	}
+	defer inf.Close()
+	tx := inf.Tx.Tx
+
+	cdn, err := validateRequest(r, inf.Version)
+	if err != nil {
+		api.HandleErr(w, r, tx, http.StatusBadRequest, err, nil)
+		return
+	}
+
+	var exists bool
+	if err = tx.QueryRow("SELECT EXISTS(SELECT id from cdn where name = $1)", cdn.Name).Scan(&exists); err != nil {
+		api.HandleErr(w, r, tx, http.StatusInternalServerError, nil, fmt.Errorf("error: %w, when checking if cdn with name %s exists", err, cdn.Name))
+		return
+	}
+	if exists {
+		api.HandleErr(w, r, tx, http.StatusBadRequest, fmt.Errorf("cdn name '%s' already exists.", cdn.Name), nil)
+		return
+	}
+
+	cdn.DomainName = strings.ToLower(cdn.DomainName)
+
+	rows, err := inf.Tx.NamedQuery(insertQuery(inf.Version), cdn)
+	if err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
+			api.HandleErr(w, r, tx, http.StatusInternalServerError, fmt.Errorf("error: %w in creating cdn with name: %s", err, cdn.Name), nil)
+			return
+		}
+		usrErr, sysErr, code := api.ParseDBError(err)
+		api.HandleErr(w, r, tx, code, usrErr, sysErr)
+		return
+	}
+	if rows.Next() {
+		if err = rows.Scan(&cdn.ID, &cdn.LastUpdated); err != nil {
+			usrErr, sysErr, code := api.ParseDBError(err)
+			api.HandleErr(w, r, tx, code, usrErr, sysErr)
+			return
+		}
+	}
+
+	alerts := tc.CreateAlerts(tc.SuccessLevel, "cdn was created.")
+	w.Header().Set("Location", fmt.Sprintf("/api/%d.%d/cdns?name=%s", inf.Version.Major, inf.Version.Minor, cdn.Name))
+	api.WriteAlertsObj(w, r, http.StatusCreated, alerts, cdn)
+	return
+}
+
+func Update(w http.ResponseWriter, r *http.Request) {
```

Comment:
V5 Update Function not returning CDN ID like V4.

**V5 Response**
```{
	"alerts": [
		{
			"text": "cdn was updated.",
			"level": "success"
		}
	],
	"response": {
		"dnssecEnabled": false,
		"domainName": "test1new",
		"id": 0,
		"lastUpdated": "2023-07-31T12:20:17.617043+05:30",
		"name": "a-test11",
		"ttlOverride": 60
	}
}
```

**V4 Response**
```{
	"alerts": [
		{
			"text": "cdn was updated.",
			"level": "success"
		}
	],
	"response": {
		"dnssecEnabled": false,
		"domainName": "test1new",
		"id": 1779,
		"lastUpdated": "2023-07-31 12:20:41+05",
		"name": "a-test111",
		"ttlOverride": 60
	}
}
```

Reply:
Verified. PUT operation returning CDN ID.